ca/transport.go

```diff
@@ -135,6 +135,36 @@ func (c *MutableTLSCreds) LoadNewTLSConfig(newConfig *tls.Config) error {
 	return nil
 }
 
+// UpdateCAs updates the root CAs and client CAs of the existing TLS config in place
+func (c *MutableTLSCreds) UpdateCAs(rootCAs, clientCAs *x509.CertPool) {
+	c.Lock()
+	defer c.Unlock()
+	// TODO(cyli): on upgrade to go 1.8.x, use c.config.Clone() instead
```

Comment:
I looked for `Clone` on both branches to verify, but did not notice it's unexported in 1.7.

I remember https://github.com/docker/docker/blob/master/pkg/tlsconfig dealing with this, and I see this was back in September. I thought it happened before Go 1.7 was released. Apparently it was right after. My apologies.

Anyway, would you mind vendoring/importing `github.com/docker/docker/pkg/tlsconfig` in this PR?